Src/Base/AMReX_ParallelDescriptor.cpp

```diff
@@ -428,7 +428,7 @@ EndParallel ()
         m_mpi_ops.clear();
     }
 
-    if (!call_mpi_finalize) {
+    if (!call_mpi_finalize && m_comm != MPI_COMM_NULL) {
```

Comment:
I think we had better understand why this is needed. It does not hurt to add the additional test. But how did it get into the situation?

Reply:
I think the situation is:
- MPI is already initialized externally.
- But I do not pass the MPI communicator to `amrex::Initialize()` and rely on the default argument `MPI_COMM_WORLD` instead.

Reply:
So I agree, should fix `StartParallel` instead.

Reply:
Huh, there is an 
```
 else {
        MPI_Comm_dup(a_mpi_comm, &m_comm);
        call_mpi_finalize = 0;
    }
```
Why does that not work ... o.0